src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int NUMBER_OF_FIGURES = 10;
+
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        Figure[] figures = new Figure[NUMBER_OF_FIGURES];
+        for (int i = 0; i < NUMBER_OF_FIGURES; i++) {
+            if (i < NUMBER_OF_FIGURES / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
```

Comment:
Make sure that you don't return `null` from the `getRandomFigure` or `getDefaultFigure` methods in the `FigureSupplier` class. It's a common mistake to return `null` which can lead to `NullPointerException`. Instead, handle all possible cases explicitly.